app/src/main/java/com/github/sdp/ratemyepfl/activity/map/MapActivity.kt

```diff
@@ -2,48 +2,187 @@ package com.github.sdp.ratemyepfl.activity.map
 
 import android.Manifest
 import android.annotation.SuppressLint
+import android.content.Intent
 import android.content.pm.PackageManager
 import android.location.Location
 import android.os.Bundle
 import android.view.Menu
 import android.view.MenuItem
+import android.widget.Button
+import android.widget.ImageView
+import android.widget.TextView
 import android.widget.Toast
+import androidx.activity.viewModels
+import androidx.annotation.NonNull
 import androidx.appcompat.app.AppCompatActivity
 import androidx.core.app.ActivityCompat
 import com.github.sdp.ratemyepfl.R
+import com.github.sdp.ratemyepfl.activity.ReviewActivity
+import com.github.sdp.ratemyepfl.model.items.Restaurant
+import com.github.sdp.ratemyepfl.model.items.RestaurantItem
+import com.github.sdp.ratemyepfl.utils.MapActivityUtils
 import com.github.sdp.ratemyepfl.utils.PermissionUtils.isPermissionGranted
+import com.github.sdp.ratemyepfl.viewmodel.RestaurantListViewModel
+import com.google.android.gms.maps.CameraUpdateFactory
 import com.google.android.gms.maps.GoogleMap
 import com.google.android.gms.maps.OnMapReadyCallback
 import com.google.android.gms.maps.SupportMapFragment
+import com.google.android.gms.maps.model.*
+import com.google.maps.android.clustering.Cluster
+import com.google.maps.android.clustering.ClusterManager
+import com.google.maps.android.clustering.view.DefaultClusterRenderer
+import com.sothree.slidinguppanel.SlidingUpPanelLayout
 import dagger.hilt.android.AndroidEntryPoint
 
-
 @AndroidEntryPoint
 class MapActivity : AppCompatActivity(), GoogleMap.OnMyLocationButtonClickListener,

```

Comment:
I see that the map does not use any feature of the Activity, and this is excellent for a refactor into a fragment. In the main activity, we use the nav_graph_main to manage the navigation, and there is this map fragment filler that is still unused (redirects to this activity). Do you think that you can change this activity into fragments, and replace the current "MapFragment" with it? Otherwise, we will do it during the next sprint :)

Reply:
You are utterly right, I will make RestaurantItem a decorator of Restaurant, it makes much more sense!

And regarding the fragment part, for sure I can refactor it into a fragment, I am going to handle that :)